### ISSUE TITLE
Freetype 2.7.0 is out. Will be there infinality patch ?

### ISSUE THREAD

**Issue by perfect7gentleman** [bohoomil/fontconfig-ultimate#171]
Or is this the end of infinality?

**Comment by Brottweiler** [comment]
Why would it be the end?

**Comment by perfect7gentleman** [comment]
I have a bad feeling

**Comment by archfan** [comment]
Did they just merge some of the infinality patches upstream?

http://phoronix.com/scan.php?page=news_item&px=FreeType-2.7-Rendering

**Comment by CamilleScholtz** [comment]
> http://phoronix.com/scan.php?page=news_item&px=FreeType-2.7-Rendering

Copied a reddit comment about this of mine:

> I'm pretty sure they enabled subpixelrendering by default in 2.6.4, I did a comparison of it between vanilla (https://punpun.xyz/9To7.png), and 2.6.3 patched with infinality (https://punpun.xyz/Pg6J.png).

**Comment by rumpelsepp** [comment]
2.7 is out; @bohoomil

**Comment by perfect7gentleman** [comment]
@rumpelsepp, 2.7 is still developing.

**Comment by rumpelsepp** [comment]
Oh, sry. I just read this http://phoronix.com/scan.php?page=news_item&px=FreeType-2.7-Rendering and I confused it with a release announcement.

On July 19, 2016 8:45:00 AM GMT+02:00, Perfect Gentleman notifications@github.com wrote:

> @rumpelsepp, 2.7 is still developing.
> 
> ---
> 
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub:
> https://github.com/bohoomil/fontconfig-ultimate/issues/171#issuecomment-233545870

## 

Sent from my Android device with K-9 Mail. Please excuse my brevity.

**Comment by afettouhi** [comment]
Is this repo dead in th water? The packages are already behind what is in Arch Linux.

**Comment by rumpelsepp** [comment]
I hope @bohoomil is doing well. He seems to be very quiet the last weeks...

**Comment by archfan** [comment]
Let's hope for the best. Maybe he's just caught up with life.

**Comment by archfan** [comment]
Just took the time to update the Infinality Bundle for FreeType 2.6.5 and Fontconfig 2.12.1.

It compiles fine and looks as brilliant as before. ;)

Please test: https://github.com/archfan/infinality_bundle

**Comment by CamilleScholtz** [comment]
@archfan Just compiled it and I can confirm everything still works and looks the same (so far :))!

**Comment by vdcow** [comment]
FYI: 2.6.5 just reverts changes made in 2.6.4. So 2.6.5 is almost the same as 2.6.3 (there was additional fix in build for Mac OS).

**Comment by CamilleScholtz** [comment]
This can be renamed to `Freetype 2.7.0 is out. Will be there infinality patch?` :)

**Comment by archfan** [comment]
JUST HAVE FAITH.

**Comment by archfan** [comment]
It's done.

https://github.com/archfan/infinality_bundle/blob/master/01_freetype2-iu/0002-infinality-2.7-2016.09.09.patch

I've added the subpixel hinting patch from Arch. Please test if changing the interpreter version in /etc/profile.d/freetype2.sh still works.

**Comment by SolarAquarion** [comment]
still the fonts still need to be updated

**Comment by afettouhi** [comment]
Also the repo needs to be updated if we get the patch.

**Comment by gothmog123** [comment]
Hi doesn anyone have any info on what is going on with Bohoomil? Is the arch repo going to be updated?

**Comment by slokhorst** [comment]
So is there any difference between FT 2.6.3+Infinality patch and FT 2.7.0 with [`truetype:interpreter-version=38` ("Infinality" mode)](https://wiki.archlinux.org/index.php/Font_configuration#Subpixel_rendering)?

(I don't know what settings were used in the previous comparisons...)

**Comment by vdcow** [comment]
@slokhorst Yes, there are some differences because Infinality patch includes some additional improvements not included to vanila freetype.

**Comment by rumpelsepp** [comment]
> some additional improvements not included to vanila freetype.

What are they? I am thinking of dropping infinatliy freetype in favor of upstream 2.7 and keep the fontconfig from here, as I mentioned here: https://github.com/bohoomil/fontconfig-ultimate/issues/183#issuecomment-258592438

**Comment by vdcow** [comment]
@rumpelsepp 

> What are they?

Unfortunately bohoomil removed original description for additional features from infinality-settings.sh but I am still tracking them in my package for Alt Linux

http://git.altlinux.org/people/cow/packages/?p=libfreetype-infinality.git;a=blob;f=infinality-settings.sh;h=6c02ec4084d08efc9b86d456bb0ef65b4a8d754c;hb=sisyphus

Also in this repo you can find actual patches for 2.7.

**Comment by julroy67** [comment]
Sadly @bohoomil doesn't respond to mails and hasn't been active on Github for over 5 months. So I'm thinking it's time to fork, because I really love the font substitution stuff.

If someone wants to fork, I'm ready to help get it up running.

**Comment by gothmog123** [comment]
Yeah just fontconfig-infinality works great all by itself. It's one package, I'm sure the gurus out there can keep it going.

**Comment by SammysHP** [comment]
Here's another blog post: https://www.freetype.org/freetype2/docs/subpixel-hinting.html

**Comment by CamilleScholtz** [comment]
I made some comparisons between 2.7 with patched infinality patches, and vanilla 2.7, and infinality _still_ looked better, especially kanji. so infinality is probably doing some more magic.

**Comment by SammysHP** [comment]
Here are some comparisons between _freetype2-infinality-ultimate_ and _freetype2_ with different values of `FREETYPE_PROPERTIES`.

[freetype2-infinality-ultimate](https://cloud.githubusercontent.com/assets/922874/20193140/fb007fcc-a78c-11e6-81c5-5d6259abb338.png)
[freetype2 `FREETYPE_PROPERTIES="truetype:interpreter-version=40"` (the new stuff, default)](https://cloud.githubusercontent.com/assets/922874/20193157/1150f2c0-a78d-11e6-89a0-500b6ba4b5ca.png)
[freetype2 `FREETYPE_PROPERTIES="truetype:interpreter-version=38"` ("infinality")](https://cloud.githubusercontent.com/assets/922874/20193182/2d50803a-a78d-11e6-83b6-6209f40fdea7.png)
[freetype2 `FREETYPE_PROPERTIES="truetype:interpreter-version=35"` (the old sstuff)](https://cloud.githubusercontent.com/assets/922874/20193203/46638360-a78d-11e6-821f-552896aae602.png)

As you can see, with most fonts there is no difference. I kept using _fontconfig-infinality-ultimate_ because I don't have a good rollback plan yet to switch back if something goes wrong (or I don't like the "normal" fontconfig). Further testing necessary.

Actually upstream freetype has an improvement. Just look at the "com" in the header: with infinality-ultimate the "o" is a little bit too tall.

If you want to try installing just freetype2 you can use this command:

```
# pacman -S freetype2 --assume-installed freetype2-infinality-ultimate
```

This will satisfy the dependency of _fontconfig-infinality-ultimate_.

**edit:** Here is another comparison with Palatino Linotype (text body), Arial (headings) and Courier New (code).

[freetype2 `FREETYPE_PROPERTIES="truetype:interpreter-version=40"`](https://cloud.githubusercontent.com/assets/922874/20194646/7897a9dc-a793-11e6-8d42-e3f3f9be3752.png)
[freetype2-infinality-ultimate](https://cloud.githubusercontent.com/assets/922874/20194647/7898657a-a793-11e6-84d4-4e0fb7ed073d.png)

Pay attention to:
- letter heights in the main heading
- hinting of the "N" in the text body (freetype2 is really bad!)
- weight of the code (freetype2 extremely light)

**Comment by afettouhi** [comment]
Someone talked about forking this? I am just interested in knowing if the repo is going to get updated to 2.7.0 or if I should go back to vanilla freetype2 in the official arch linux repoes.

**Comment by rumpelsepp** [comment]
@afettouhi I would suggest to try vanilla freetype with the fontconfig from here. Maybe we could fork the fontconfig und create some AUR packages without the dependencies to this repo.

I am thinking of providing the relevant infrastructure to make at least another font pacman repo, that is being updated regularly. The config package could live in AUR IMO; the rest can IMO be vanilla freetype, since I am not interested in maintaining patch sets to code bases I do no understand.

**Comment by afettouhi** [comment]
@rumpelsepp I don't want start mixing vanilla stuff with old stuff from the repo. Fontconfig is also behind in the repo.Upstream is at 2.12.2 or something, but I am all for if someone is willing to put it all in AUR.

**Comment by rumpelsepp** [comment]
Are there any major changes with the new font config? I personally like the font substitution rules from here and the really great fonts repo.

**Comment by gothmog123** [comment]
Is it possible to just run with the font substitution from infinality sustainably? Would one just copy the config files?

**Comment by afettouhi** [comment]
@rumpelsepp Don't know if there major changers in the 2.12.x series I am just interested in getting this update with the Arch repos.

**Comment by rumpelsepp** [comment]
For the record; I have migrated back to fully upstream packages. Namely:

* freetype 2.7
* fontconfig
* cairo
* several font packages from the arch repos (deja-vu, liberation, robotomono, droid, noto-*)

I left the default and I changed my i3wm to use the mate control center settings (I can configure the font substitutions for sans, serif, and monospace with a click now... I am lazy, I know). I am really happy with the results so far. There seem to be a lot of improvements since I used the upstream packages the last time. So, I am out of infinality.

**Comment by afettouhi** [comment]
Did the same thing, so I am also out of infinality for now.

**Comment by Brottweiler** [comment]
I've also switched to regular freetype2... Things are different, but not that bad, so it's fine.

**Comment by mattia-b89** [comment]
As already stated previously, our effort should be concentrated on font
substitution, that's the porting of font config infinality .conf files
mainly.

powered by CM

**Comment by jurf** [comment]
Yeah, I would be very interested in that, but I doubt I'll find the time in the upcoming weeks.

**Comment by julroy67** [comment]
If there is anyone interested I've started my own font collection. It uses the v40 interpreter of Freetype (but Infinality mode, aka v38, can be activated) and although I've started with bohoomil's config, I'm now using different fonts.
The repo is at https://github.com/julroy67/kagayaki

And there are some compiled packages at https://julroy67.freeboxos.fr:13476/share/h3Y-xMyPcgF6drIY/ (home server, so slow upload)

**Comment by CamilleScholtz** [comment]
@julroy67 Awesome! Can you please give me a rundown on what the patches actually add over vanilla cairo/fontconfig/freetype if possible?

**Comment by mattia-b89** [comment]
@julroy67, Yes would be great if you keep a changelog

**Comment by mattia-b89** [comment]
Does anyone tried infinality from AUR?
Especially, do you note differences with fontconfig from AUR?

**Comment by vimsucks** [comment]
@mattia-b89 AUR package [freetype2-infinality](https://aur.archlinux.org/packages/freetype2-infinality) did not work for me. 
FT_Get_Var_Blend_Coordinates error still exists.

**Comment by Brottweiler** [comment]
Probably want to use https://wiki.archlinux.org/index.php/Unofficial_user_repositories#infinality-bundle instead of AUR, maybe.

**Comment by SammysHP** [comment]
The old version of freetype2-infinality won't work anymore because of the harfbuzz update. Either switch to the official version of freetype2 or port all infinality patches to the latest version of freetype2.

**Comment by jurf** [comment]
I’ve made font meta packages to ease the migration from Infinality if anyone’s interested, you can find them on the AUR as[`fonts-meta-base`](https://aur.archlinux.org/packages/fonts-meta-base/) and [`fonts-meta-extended-lt`](https://aur.archlinux.org/packages/fonts-meta-extended-lt/).

**Comment by mattia-b89** [comment]
great start!

**Comment by liaowang11** [comment]
Tried vanilla freetype2 with infinality mode, fonts blurry. infinality-ultimate still makes fonts much better. Rollbacked, I think I'll just stop updating until infinality is ready.

**Comment by mrksr** [comment]
I was successful in getting nice rendering again by following the guide in this gist by @cryzed: https://gist.github.com/cryzed/e002e7057435f02cc7894b9e748c5671

It also includes the packages created by @DoctorJellyface. Should be a good place to start.

**Comment by RJVB** [comment]
I'm jumping into discussion just now and still a bit lost.

Am I correct that @archfan has been keeping a fork up-to-date for Freetype >= 2.7 and that the existing other patches and things from bohoomil's repo still work for building from source?
And that much of the ongoing discussion here is about getting things to work on Arch, so not really relevant for other systems? (I maintain a PPA for Ubuntu 14.04 and a set of MacPorts ports.)

**Comment by CamilleScholtz** [comment]
@RJVB Yup, I also provide Pkgfile's for CRUX, and I'm like 90% sure @archfan makes his own PKGBUILDs for Arch as well.

I don't see how it isn't relevant to other systems, there just aren't any packages for them yet.

**Comment by RJVB** [comment]
I think what I was trying to ask was something like "can I just grab archfan's" updated Freetype2 patches and use them with my own packaging stuff and Freetype 2.7.1? Or do I have to read through the whole thread because I'm *bound* to run into trouble if I don't?

Also, who actually maintains the patch set?

**Comment by CamilleScholtz** [comment]
It used to be @bohoomil, but he has been inactive lately, now @archfan forked it.

There isn't really much of a difference between the two patchsets besides the version. But I still suggest that you read through the old infinality PKGBUILD, and the PKGBUILDs created by @archfan just to see the differences. Since PKGBUILDs are basically build instructions in BASH that shouldn't be too hard.

Also since Ubuntu is a binary distro and has some old libraries you might run into incompatibilities, but I don't know. For example from the looks of this page it seems they are still on 2.6.3: https://launchpad.net/ubuntu/+source/freetype

**Comment by RJVB** [comment]
OK, so it's archfan's repo I'll have to follow, thanks. I've been doing my own packaging based on bohoomil's instructions for years now, so if the patchset changes only concern adapting to Freetype 2.7 I should be fine.

Freetype is written in C meaning there are no issues with runtimes, and it seems 2.7 ABI is backwards compatible with the 2.6 ABI. I grabbed the stock libfreetype2.so.6.13.0 file from my KaOS partition, put it alongside my Infinality libfreetype2.so.6.12.5 in Ubuntu's /usr/lib/x86_64 directory. Then I symlinked libfreetype2.so.6 to the newer library, and ran a few tests. No issues.

Surprisingly, not even a significantly decreased font quality, maybe an even slightly *better* quality, somewhat darker/sharper. Maybe I *should* read the whole thread here, because if indeed the main quality improvement comes from the fontconfig patches and the .conf files that means I might have 1 less package to maintain ...
EDIT: forget it, a tricky font like Noverase looks a lot better WITH the Infinality patches.

**Comment by RJVB** [comment]
I have now built and tested Freetype 2.7.1 with archfan's latest patch, under XQuartz. Without changing anything to my environment I must say I don't see any difference with Freetype+Infinality 2.6.5, regardless of how I set the new FREETYPE_PROPERTIES env. variable. So far so good.

Allow me to ask: is there a package source somewhere that I can add to my KaOS test install? I'd rather keep that one as "stock" as possible.
